usbl_gazebo/CMakeLists.txt

```diff
@@ -1,17 +1,11 @@
 cmake_minimum_required(VERSION 3.0.2)
 project(usbl_gazebo)
 
-## Compile as C++11, supported in ROS Kinetic and newer
-# If running on focal, will be using Noetic + Gazebo11, which requires c++17
-execute_process(COMMAND lsb_release -cs
-  OUTPUT_VARIABLE RELEASE_CODENAME
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-if(RELEASE_CODENAME MATCHES "focal")
-  set(CMAKE_CXX_STANDARD 17)
-  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+if(NOT "${CMAKE_VERSION}" VERSION_LESS "3.16")
```

Comment:
I like determining c++17 vs c++11 by CMAKE_VERSION rather than the os, since it's more future proof. I guess it has the potential to change behavior for somebody on Melodic if they upgrade (but the 18.04 debs only have 3.10). I don't think this should be a problem, but I defer to @crvogt for that.

Reply:
Another benefit of using `CMAKE_VERSION` is that it works on more than just the standard Ubuntu distribution. It can also work on other Debian derivatives as well as Fedora.